de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/project/SharedResourcesManager.java

```diff
@@ -195,126 +195,194 @@ private void handleFileRecovery(FileActivity activity) throws IOException {
         } finally {
             /*
              * always reset Jupiter or we will get into trouble because the
-             * vector time is already reseted on the host
+             * vector time has already been reset on the host
              */
             sarosSession.getConcurrentDocumentClient().reset(path);
         }
     }
 
-    private void handleFileMove(FileActivity activity) throws IOException {
+    /**
+     * Applies the given move FileActivity. Subsequently cleans up the
+     * EditorPool and AnnotationManager for the moved file if necessary.
+     *
+     * @param activity the move activity to execute
+     * @throws IOException if the creation of the new file or the deletion of
+     *                     the old file fails
+     */
+    private void handleFileMove(
+        @NotNull
+            FileActivity activity) throws IOException {
+
         SPath oldPath = activity.getOldPath();
         SPath newPath = activity.getPath();
 
-        IntelliJProjectImpl oldProject =
-            (IntelliJProjectImpl) oldPath.getProject();
-        IntelliJProjectImpl newProject =
-            (IntelliJProjectImpl) newPath.getProject();
+        IFile oldFile = oldPath.getFile();
+        IFile newFile = newPath.getFile();
+
+        if (!oldFile.exists()) {
+            LOG.warn("Could not move file " + oldFile + " as it does not exist."
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal " as it does not exist." 3 times. ([squid:S1192](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1192))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=277&issues=AWagwjc6lHK86HcoyTJG&open=AWagwjc6lHK86HcoyTJG)

[ISSUE_KEY]: AWagwjc6lHK86HcoyTJG